src/machines/modelingMachine.ts

```diff
@@ -952,3 +951,65 @@ export const modelingMachine = createMachine(
     },
   }
 )
+
+function getSketchMetadataFromPathToNode(
```

Comment:
There was a lot of logic that could be reused between the `set sketch metadata` and `set sketchMetadata from pathToNode` actions that have been put into this function.